### ISSUE TITLE
Fix request: AvoidInfiniteRecursion - same (interface) method on new object or cast field or simple other type

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#505]
```java
    public InputStream execute(Context context) {
        InputStream stream = new ResourceCmd(id, name).execute(context); // <-- false positive
        return stream;
    }

    public InputStream execute2(Context context) {
        InputStream stream = ((Other)field).execute2(context); // <-- false positive
        return stream;
    }

    public String sourceUrl(String url) {
        try {
            return sourceUrl(new URL(url)); // <-- false positive
        } catch (MalformedURLException e) {
            throw new IllegalArgumentException("malformed url: " + url, e);
        }
    }
```

**Comment by jborgers** [comment]
the last one was already fixed
